### PR TITLE
Fix for #1239

### LINKED ISSUE #1239: invalid registered toolchain '@rules_foreign_cc//toolchains:preinstalled_nmake_toolchain': No repository visible as '@rules_foreign_cc' from main repository
I have simple Bazel module
MODULE.bazel:
```starlark
module(
    name = "libxml2",
    version = "2.13.3",
    compatibility_level = 1,
)
bazel_dep(name = "rules_foreign_cc", version = "0.11.1")
```
BUILD.bazel:
```starlark
""" Builds libxml2.
"""

load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")

filegroup(
    name = "srcs",
    srcs = glob(["**"]),
)

cache_entries = {
    "CMAKE_POSITION_INDEPENDENT_CODE": "ON",
    "BUILD_SHARED_LIBS": "OFF",
    #libxml2 specific options.
    "LIBXML2_WITH_PYTHON": "OFF",
    "LIBXML2_WITH_LZMA": "OFF",
    "LIBXML2_WITH_TESTS": "OFF",
    "LIBXML2_WITH_PROGRAMS": "OFF",
}

cmake(
    name = "libxml2",
    env = {"CMAKE_BUILD_TYPE": "Release"},
    lib_source = ":srcs",
    cache_entries = cache_entries,
    out_include_dir = "include/libxml2",
    visibility = ["//visibility:public"],
)
```
This module compiles successfully on Linux and macOS. But on Windows the following error occurs:
```powershell
bazel --output_user_root=C:/b build --show_progress_rate_limit=5 --curses=yes --color=yes --terminal_columns=143 --show_timestamps --verbose_failures --jobs=30 --announce_rc --experimental_repository_cache_hardlinks --disk_cache= --remote_cache=remotebuildexecution.googleapis.com --remote_instance_name=projects/bazel-untrusted/instances/default_instance --google_default_credentials --bes_backend=buildeventservice.googleapis.com --bes_timeout=360s --project_id=bazel-untrusted --remote_timeout=60 --remote_max_connections=200 --remote_default_platform_properties=properties:{name:"cache-silo-key" value:"1d5ec805093884c76b310cf26aeb82ee568b9359f72bef33e0bc82eff4360c80"} --remote_download_toplevel --test_env=LocalAppData --test_env=BAZELISK_USER_AGENT --test_env=USE_BAZEL_VERSION -- @libxml2//:libxml2
(16:48:01) WARNING: Option 'project_id' is deprecated: Use --bes_instance_name instead
(16:48:01) WARNING: Option 'remote_default_platform_properties' is deprecated: --remote_default_platform_properties has been deprecated in favor of --remote_default_exec_properties.
(16:48:01) INFO: Invocation ID: 0f7cc2aa-16f9-4098-a3fb-4d089a40eca6
(16:48:01) INFO: Options provided by the client:
  Inherited 'common' options: --isatty=0 --terminal_columns=80
(16:48:01) INFO: Options provided by the client:
  'build' options: --python_path=C:/python3/python.exe
(16:48:01) INFO: Reading rc options for 'build' from c:\b\bk-windows-326v\bazel-org-repo-root\.bazelrc:
  'build' options: --experimental_enable_bzlmod --registry=file:///c:/b/bk-windows-326v/bazel/bcr-presubmit
(16:48:01) INFO: Current date is 2024-07-30
(16:48:01) WARNING: Option 'project_id' is deprecated: Use --bes_instance_name instead
(16:48:01) WARNING: Option 'remote_default_platform_properties' is deprecated: --remote_default_platform_properties has been deprecated in favor of --remote_default_exec_properties.
(16:48:12) ERROR: C:/b/wd2wermq/external/rules_foreign_cc~/toolchains/private/BUILD.bazel:158:15: While resolving toolchains for target @@rules_foreign_cc~//toolchains/private:pkgconfig_tool_msvc_build_ (538d53d): invalid registered toolchain '@rules_foreign_cc//toolchains:preinstalled_nmake_toolchain': No repository visible as '@rules_foreign_cc' from main repository
(16:48:12) ERROR: Analysis of target '@@libxml2~//:libxml2' failed; build aborted: Analysis failed
(16:48:12) INFO: Elapsed time: 11.979s, Critical Path: 0.05s
(16:48:12) INFO: 1 process: 1 internal.
(16:48:12) ERROR: Build did NOT complete successfully
(16:48:12) FAILED:
    Fetching repository @@rules_foreign_cc~~tools~cmake-3.23.2-windows-x86_64; starting 5s
    Fetching C:/b/wd2wermq/external/rules_foreign_cc~~tools~cmake-3.23.2-windows-x86_64; Extracting cmake-3.23.2-windows-x86_64.zip 4s
bazel build failed with exit code 1
🚨 Error: The command exited with status 1
user command error: exit status 1
```
Full logs are available here - https://buildkite.com/bazel/bcr-presubmit/builds/6949#0191048a-50a6-4826-8e5c-68d888cca6f1
Related discussion in BCR - https://github.com/bazelbuild/bazel-central-registry/pull/2512

### PR DESCRIPTION
**Pull request by dmitry-j-mikhin**
Closes #1239

### CODE REVIEW

**Review by meteorcloudy — APPROVED** [review]
Thanks! I didn't know rules_cc was a dev_dependency.

**Comment by meteorcloudy** [comment]
@jsharpe Please take a final look.

File: MODULE.bazel [hunk 1]
```diff
@@ -10,12 +10,12 @@ bazel_dep(name = "bazel_features", version = "1.15.0")
 bazel_dep(name = "bazel_skylib", version = "1.3.0")
 bazel_dep(name = "platforms", version = "0.0.5")
 bazel_dep(name = "rules_python", version = "0.23.1")
+bazel_dep(name = "rules_cc", version = "0.0.9")
```
Comment (jsharpe) [inline MODULE.bazel:13]: `rules_cc` shouldn't be required at all - I'll put in a separate PR to rectify this

Comment (jsharpe) [reply MODULE.bazel:13]: Ah rules_cc as a dev_dependency is only present to satisfy the rbe configuration - that package needs some work to fix it up properly..

Comment (dmitry-j-mikhin) [reply MODULE.bazel:13]: @jsharpe without it I have the following error on Windows:
```
The repository '@@[unknown repo 'rules_cc' requested from @@rules_foreign_cc~~tools~glib_dev]' could not be resolved: No repository visible as '@rules_cc' from repository '@@rules_foreign_cc~~tools~glib_dev'. and referenced by '@@rules_foreign_cc~//toolchains/private:pkgconfig_tool_msvc_build_'
```
So it was added to fix the error above.

Comment (jsharpe) [reply MODULE.bazel:13]: I know but the solution isn't to start depending on `rules_cc`.  I'm addressing this in https://github.com/bazelbuild/rules_foreign_cc/pull/1269 instead.

Comment (dmitry-j-mikhin) [reply MODULE.bazel:13]: Okay, got it.

Comment (jsharpe) [reply MODULE.bazel:13]: I've merged 1269 now so if you remove this files changes from this PR then we can land this fix.

Comment (dmitry-j-mikhin) [reply MODULE.bazel]: @jsharpe I removed it.

**Review by jsharpe — APPROVED** [review]

**Comment by dmitry-j-mikhin** [comment]
I created a separate PR for the APR issue - https://github.com/bazelbuild/rules_foreign_cc/pull/1273 and rebased the current branch onto it to make CI/CD green again.

**Comment by jsharpe** [comment]
> I created a separate PR for the APR issue - #1273 and rebased the current branch onto it to make CI/CD green again.

Thank you!

**Pull request merged**